ethdb/privateapi/ethbackend.go

```diff
@@ -262,7 +241,7 @@ func (s *EthBackendServer) EngineExecutePayloadV1(ctx context.Context, req *type
 	log.Info("Received Payload from beacon-chain", "hash", blockHash)
 	// Send the block over
 	s.numberSent = req.BlockNumber
-	s.reverseDownloadCh <- *types.NewBlock(&header, transactions, nil, nil)
```

Comment:
We need the entire block to be able to execute it, not just the reader.

Reply:
my thought process is this, we first use the headers and use it to do stage headers, after it we re-download the body and execute the subsequent stage, result: we obtain the merge by changing the minimum amount of code. yes, this add a little bit of overhead, but as a first prototype i think we can do this so that we dont add too much complexity right away.

Reply:
OK, I guess we can do it for the first prototype.